modules/macos_auto_puppet/files/auto-puppet.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Purpose: Bootstrap a macOS 14+ host from post-install (or image) to a complete Puppet run.
+
+export LANG=en_US.UTF-8
+export PATH="/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/puppetlabs/bin"
+
+# Puppet configuration
+LOCAL_PUPPET_REPO="/Users/relops/Desktop/puppet/ronin_puppet"
+PUPPET_ROLE_FILE="/etc/puppet_role"
+PUPPET_BIN="/opt/puppetlabs/bin/puppet"
+FACTER_BIN="/opt/puppetlabs/bin/facter"
+GIT_REPO_URL="https://github.com/mozilla-platform-ops/ronin_puppet.git"
+GIT_BRANCH="auto_puppet_v2"
+
+# Vault configuration
+export VAULT_ADDR="http://127.0.0.1:8200"
+VAULT_TOKEN="$(cat /etc/vault_token 2>/dev/null)"
+export VAULT_TOKEN
+
+# Fail function
+fail() {
+    echo "${@}"
+    exit 1
+}
+
+# Ensure Puppet Role is Set
+if [ -f "$PUPPET_ROLE_FILE" ]; then
+    ROLE=$(<"$PUPPET_ROLE_FILE")
+else
+    fail "Failed to find Puppet role file $PUPPET_ROLE_FILE"
+fi
+
+# Ensure Puppet & Facter are Installed
+if [ ! -x "$PUPPET_BIN" ]; then
+    fail "Puppet is missing or not executable."
+fi
+
+if [ ! -x "$FACTER_BIN" ]; then
+    fail "Facter is missing or not executable."
+fi
+
+# Remove existing Puppet repository and re-clone it
+rm -rf "$LOCAL_PUPPET_REPO"
+git clone --branch "$GIT_BRANCH" "$GIT_REPO_URL" "$LOCAL_PUPPET_REPO" || fail "Failed to clone Puppet repository"
```

Comment:
A bit wasteful, but i guess it avoids issues with the checkout dir (PUPPET_ENV) that I had to deal with on linux.

Reply:
We were just talking about DC network usage and the ticket about throttling came up. 

Github has been pretty good about not throttling us, but if they did start this would become a problem. Since it's in the critical flow of the worker, no tasks would be able to run.

The linux script (I think) has solved the issue with updating existing clients.